platform/podjs/bin/genPodjs.js

```diff
@@ -0,0 +1,75 @@
+const MagicString = require("magic-string");
+const fs = require("fs");
+
+function replaceManifestData(code, manifestData) {
+    const magicString = new MagicString(code);
+    const pattern = new RegExp("window.manifestData", "g");
+
+    let hasReplacements = false;
+    let match;
+    let start;
+    let end;
+    let replacement;
+
+    while ((match = pattern.exec(code))) {
+        hasReplacements = true;
+
+        start = match.index;
```

Comment:
Why is this variable defined outside the loop? It's simply a constant. Same for `match`, `end` and even `replacement`

Reply:
cause it's inside the while... we should be ignorant to how many times the pattern is matched.

Reply:
Let me rephrase that. Is there a difference between this and doing 

```
const start = match.index?
```

or even 

```js
const end = match.index + match[0].length
```

?